docs/configuring-playbook-user-verification-service.md

```diff
@@ -0,0 +1,116 @@
+
+In order to use UVS, an admin token for the configured homeserver must be supplied. For now this means configuring Synapse and creating the token before installing UVS.
+
+## Enable
+
+[Matrix User Verification Service](https://github.com/matrix-org/matrix-user-verification-service) installation is disabled by default unless required by Jitsi (see group_vars/matrix_servers). 
+You can enable it in your configuration file (`inventory/host_vars/matrix.<your-domain>/vars.yml`):
+
+```yaml
+matrix_user_verification_service_enabled: true
+```
+
+## Configuration
+
+The only required configuration variable is `matrix_user_verification_service_uvs_access_token` (see below).
+
+For a list of all configuration options see the role defaults [`roles/matrix-user-verification-service/defaults/main.yml`](../roles/custom/matrix-user-verification-service/defaults/main.yml).
+But be aware of all the plugging happening in `group_vars/matrix_servers`.
+
+In the default configuration, the UVS Server is only reachable via the docker network, which is fine if e.g. Jitsi is also running in a container on the host.
+However, it is possible to expose UVS via setting `matrix_user_verification_service_container_http_host_bind_port`. Be aware that the normally used port (3000) may collide with Grafana.
+
+### Access token
+
+The Synapse Access Token is used to verify RoomMembership and PowerLevel against the configured homeserver_url (which is plugged in group_vars).
+
+We recommend that you create a dedicated Matrix user for uvs (`uvs` is a good username).
+Follow our [Registering users](registering-users.md) guide to register a user with administration privileges.
+
+You are required to specify an access token (belonging to this new user) for UVS to work.
+To get an access token for the UVS user, you can follow the documentation on [how to do obtain an access token](obtaining-access-tokens.md).
+
+**Access tokens are sensitive information. Do not include them in any bug reports, messages, or logs. Do not share the access token with anyone.**
+
+```yaml
+matrix_user_verification_service_uvs_access_token: "YOUR ACCESS TOKEN HERE"
+```
+
+### (Optional) Auth Token
+
+It is possible to set an API Auth Token to restrict access to the UVS. If this is set, anyone making a request to UVS must provide it via the header "Authorization: Bearer TOKEN"
+
+By default, the token will be derived from `matrix_homeserver_generic_secret_key` in `group_vars/matrix_servers`.
+To set your own Token, simply put the following in your host_vars.
+
+```yaml
+matrix_user_verification_service_uvs_auth_token: "TOKEN"
+```
+
+In case Jitsi is also managed by this collection and 'matrix' authentication in Jitsi is enabled, this collection will automatically configure Jitsi to use the configured auth token.
+
+### (Optional) Federation
+
+In theory (however currently untested), UVS can handle federation. Simply set:
+
+```yaml
+matrix_user_verification_service_uvs_openid_verify_server_name: ~
```

Comment:
This `matrix_user_verification_service_uvs_openid_verify_server_name` is somewhat messy.

Looks like it's not defined in `defaults/main.yml`, yet... `group_vars/matrix_servers` defines it, so it's always defined and various `matrix_user_verification_service_uvs_openid_verify_server_name is defined` checks will always pass.

`group_vars/matrix_servers` defines it in via some `if` statement, which has the chance to set `matrix_user_verification_service_uvs_openid_verify_server_name` to an empty string, but can never really set it to `null` due to how Ansible/Jinja if statements work.

Then there's this documentation here telling people to hardcode it as `null`. It would work, but.. the whole thing is messy.. undefined in the role, checks for undefined which will pretty much never work, possibly set to empty string, telling people about `null`.

It's better if `matrix_user_verification_service_uvs_openid_verify_server_name` is defined in `defaults/main.yml` and all this documentation is there (at least). We can then still keep the documentation here as well, but we can at least drop the `is defined` checks.

I think you can probably even make drop the `null` thing and hardcode `matrix_user_verification_service_uvs_openid_verify_server_name: "{{ matrix_domain }}"` in `defaults/main.yml` and clean up `group_vars/matrix_servers`.

Then your checks can be.. `if matrix_user_verification_service_uvs_openid_verify_server_name | length > 0` or `if matrix_user_verification_service_uvs_openid_verify_server_name != ''`